eosdis-nasa/earthdata-pub-api
# Code Generation Backend Support

# Description

Creates backend support for code generation. This is accomplished through a new "code" table  and sql queries against that table.

## Spec

See Ticket: https://bugs.earthdata.nasa.gov/browse/EDPUB-1373

---

## Validation

1. Make sure all merge request checks have passed (CI/CD).
2. Pull related branches locally.
3. Log into the dashboard and retrieve a bearer token
4. Navigate to the swagger and authorize the page with the bearer token from 3
5. Scroll down to the `/api/data/submission/validateCode/{code}` endpoint
6. Click try me out and enter a UUID4 value into the token box
7. Verify the return shows `{"isValid": false}`
8. In your database viewer of choice add an entry to the code table using a submission id in your database. 
E.X
```
INSERT INTO "code"(submission_id, daac_id)
Values('614ae1fb-c2f6-4135-be83-499b94388177', '15df4fda-ed0d-417f-9124-558fb5e5b561')
RETURNING *
```
9. Copy the returned code value and repeat step 6.
10. Verify that the return shows `{"isValid": true}`


---

## Change Log

* Adds backend support for code generation
